### ISSUE TITLE
Cluster partition Actor PID transfer issue (duplicate grains)

### ISSUE THREAD

**Issue by DenizPiri** [AsynkronIT/protoactor-dotnet#518]
## Expected Behavior
When adding new nodes to a cluster, a single grain name should always map to a consistent PID.

## Actual Behavior
After N number of nodes, after different nodes of the cluster are requesting PIDs, they start getting different PIDs.

The actual problem is masked by the PidCache. When that is disabled, the problem transforms into PID changing as new nodes are added to the cluster.

I determined the problem to be related to Actors in a partition not being transferred correctly. When I dived deeper, I figured out that the problem is caused by a node being added to the MemberList even if the Alive property is false (consul provider actively uses this property). Together with that addition, code tries to determine which actors should be transferred, however, inside Rendezvous.GetNode() the Alive property of members are checked and skipped if they are not Alive.

So basically new node gets added, it is not Alive, no transfer gets performed. When the node later comes alive, code doesn't check for old.Alive != new.Alive state and transfer the actor PIDs.

Adding this at the top of the MemberList.UpdateAndNotify() function fixes the problem.

```
if (old != null && old.Alive == false)
    old = null;
if (@new != null && @new.Alive == false)
    @new = null;
```

However, I am thinking of making a better solution by adding a new MemberAlivenessUpdated message and triggering actor PID transfers only in that case. I will try to make a pull request soon.

**Comment by DenizPiri** [comment]
https://github.com/AsynkronIT/protoactor-dotnet/pull/519 is merged and fixes this issue. Closing this. 🎉
